### PR TITLE
Count splits for unstarted source nodes as queuedDrivers

### PR DESCRIPTION
**Pull request by haozhun**
Fixes bug 2 in https://github.com/prestodb/presto/issues/11253

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/operator/PipelineContext.java [hunk 1]
```diff
@@ -324,11 +331,12 @@ public PipelineStats getPipelineStats()
             lastExecutionEndTime.compareAndSet(null, now);
         }
 
+        int completedDrivers = this.completedDrivers.get();
         List<DriverContext> driverContexts = ImmutableList.copyOf(this.drivers);
-        PipelineStatus pipelineStatus = getPipelineStatus(driverContexts.iterator(), partitioned);
+        int totalSplits = this.totalSplits.get();
```
Comment (dain) [inline presto-main/src/main/java/com/facebook/presto/operator/PipelineContext.java]: maybe move this one line up, so there is less lag between the two readings

File: presto-main/src/main/java/com/facebook/presto/operator/PipelineContext.java [hunk 2]
```diff
@@ -492,6 +507,18 @@ else if (driverContext.isFullyBlocked()) {
             }
         }
 
+        int queuedDrivers;
+        if (partitioned) {
+            queuedDrivers = totalSplits - runningDrivers - blockedDrivers - completedDrivers;
+            if (queuedDrivers < 0) {
```
Comment (dain) [inline presto-main/src/main/java/com/facebook/presto/operator/PipelineContext.java]: how about `if (queuedDrivers < physicallyQueuedDrivers) queuedDrivers = physicallyQueuedDrivers`?

Comment (haozhun) [reply presto-main/src/main/java/com/facebook/presto/operator/PipelineContext.java]: I think that's a good idea.

I do have a concern about hiding bugs. But I think I have this problem here regardless of this change anyways.

Comment (dain) [reply presto-main/src/main/java/com/facebook/presto/operator/PipelineContext.java]: I'm more concerned about observation skew due to the lack of a snapshot.  I think that this provides a better answer than 0.

File: presto-main/src/main/java/com/facebook/presto/execution/SqlTaskExecution.java [hunk 3]
```diff
@@ -354,6 +354,7 @@ private void mergeIntoPendingSplits(PlanNodeId planNodeId, Set<ScheduledSplit> s
         DriverSplitRunnerFactory partitionedDriverFactory = driverRunnerFactoriesWithSplitLifeCycle.get(planNodeId);
```
[the remark this answers is not in the document]
Comment (haozhun) [reply presto-main/src/main/java/com/facebook/presto/execution/SqlTaskExecution.java]: Here, it mean "it's a partitioned split" (i.e. each split will instantiate a different driver)

File: presto-main/src/main/java/com/facebook/presto/execution/SqlTaskExecution.java [hunk 4]
```diff
@@ -354,6 +354,7 @@ private void mergeIntoPendingSplits(PlanNodeId planNodeId, Set<ScheduledSplit> s
...
         PendingSplitsForPlanNode pendingSplitsForPlanNode = pendingSplitsByPlanNode.get(planNodeId);
 
+        partitionedDriverFactory.splitsAdded(scheduledSplits.size());
```
[the remark this answers is not in the document]
Comment (haozhun) [reply presto-main/src/main/java/com/facebook/presto/execution/SqlTaskExecution.java]: See the early return in `schedulePartitionedSource(TaskSource source)` in https://github.com/prestodb/presto/commit/e8a5a9063cadb5c6b775ad22e2c7b0f31f17a317#diff-61fd8ddf279155fc3042827840e7bc9cR308

[the remark this answers is not in the document]
Comment (wenleix) [reply presto-main/src/main/java/com/facebook/presto/execution/SqlTaskExecution.java]: @haozhun : Why this a separate diff?

[the remark this answers is not in the document]
Comment (haozhun) [reply presto-main/src/main/java/com/facebook/presto/execution/SqlTaskExecution.java]: Sorry, that was intended as response to your top-level question: "why a driver for the split can be delayed instantiate now?"

**Pull request merged**